### DISCUSSION TITLE
NixOS support

### DISCUSSION THREAD

**Discussion by darren-arch** [wasp-lang/wasp#2357] [Archived]
How would I get this to work on NixOS? I get that there is a easy setup script that you normally use, however, that doesn't work on nixos because of how the OS is. It would be nice if Wasp got a nixos package, but considering that nixos may be a small amount of the user base I understand. Until then though how could I get Wasp to work with nixos? I'm also kinda dumb when it comes to using nix, but I couldn't find a way to use wasp with nix-lb and it doesn't work with steam-run.

**Comment by sodic** [comment]
Hey @darren-arch, thanks for the question.

I've never used Nix so I'm no expert here. Can you explain why the normal `curl` script doesn't work? From what I know, it should work as it normally does.

**Comment by darren-arch** [comment]
Nixos is a declarative OS, so I can't actually install things like normal. The only way I can install things is by putting the nix package name in its config file. Then when I "rebuild" the system with "nixos-rebuild switch" it installs. So anything that is a nix package is allowed on nixos, its very similar to flatpaks, however, it can share dependencies which allows it to take up much less space. But that means that you can't add things to the .local file and have it run, which sucks for flexibility.
